src/modules/TTS/splitting.js

```diff
@@ -267,21 +260,9 @@
           elements.push(splitEl);
           allElements.push(splitEl);
         });
-        // insert trailing space if there was one
-        if (wholeText[wholeText.length - 1] === " ") {
-          allElements.push(createText(" "));
-        }
-        allElements.push(
-          createElement(F, "whitespace", " ", preserveWhitespace)
-        );
+        allElements.push(createText(" "));
       } else {
-        // Don't create whitespace element where textNode is invalid
-        var isChildTextNodeInvalid = /table|thead|tbody|tfoot|tr/i.test(el.tagName);
-        if (!isChildTextNodeInvalid) {
-          allElements.push(
-            createElement(F, "whitespace", " ", preserveWhitespace)
-          );
-        }
+        allElements.push(createText(" "));
```

Comment:
Maybe conditional else if check here to add space only when needed, instead of just else.
```
} else if (wholeText.length && !contentsTrimmed.length) {
    // add back missing whitespace
    allElements.push(createText(" "));
}
```

Reply:
reverted this.

Reply:
added suggested else if